microprofile/WFLY-10711_health_check_extension.adoc

```diff
@@ -0,0 +1,82 @@
+= [WFLY-10711] Support Eclipse MicroProfile Health
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          observability,microprofile,health,openshift
+
+== Overview
+
+In order to improve Observability of WildFly on OpenShift, WildFly needs to expose health check probes
+so that the container can query whether WildFly and its deployed applications are healthy or not.
+
+This will be implemented by supporting Eclipse MicroProfile Health 1.0 for applications deployed in WildFly.
+The integration uses the smallrye-health component to provide the MicroProfile Health implementation.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-10711[WFLY-10711]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1029[EAP7-1029]
+* https://issues.jboss.org/browse/CLOUD-2730[CLOUD-2730] - Implement EAP Health Check so that it can be consumed by OpenShift Console
+* https://issues.jboss.org/browse/WFLY-10522[WFLY-10522] - MicroProfile Config extension for WildFly.
+** There is no direct relation between MicroProfile Health and MicroProfile Config but there should be consistency in their extensions (subsystem names, etc.)
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:tremes@redhat.com[Tomas Remes]
+
+=== Affected Projects or Components
+
+The implementation of Eclipse MicroProfile Health is provided by https://github.com/smallrye/smallrye-health[smallrye-health].
+
+=== Other Interested Projects
+
+* A https://github.com/jmesnil/wildfly-microprofile-health[prototype of a WildFly extension] for MicroProfile Health was developed during the development of MicroProfile Health 1.0.
+
+== Requirements
+
+* Support Eclipse MicroProfile Health 1.0 for applications deployed in WildFly.
+* Provide a HTTP endpoint for the https://github.com/eclipse/microprofile-health/blob/master/spec/src/main/asciidoc/protocol-wireformat.adoc[health wireformat]
+* The HTTP endpoint is global to the server: it must aggregate health check probes from *all deployments*.
+
+== Nice-to-Have Requirements
+
+* Support drilldown of the HTTP endpoint (e.g. to be able to query the health of individual deployment, e.g. `/health/deploymentA`)
```

Comment:
Is this in scope of planned work for WildFly 14 ?

Reply:
No, I don't plan to work on it for WildFly 14. I put it in "nice-to-have" to keep a trace of further enhancements